### PR TITLE
feat: Impl ordered Message/Event type

### PR DESCRIPTION
**Pull request by ArturVargas**
add `order` field in to **MessageToL1** and **EventContent**

### CODE REVIEW

**Comment by antiyro** [comment]
please rename struct to respectively `OrderedEvent` and `OrderedMessage`

File: src/transaction.rs [hunk 1]
```diff
@@ -467,6 +469,8 @@ pub struct Event {
 pub struct EventContent {
     pub keys: Vec<EventKey>,
     pub data: EventData,
+    /// The order of the event within the transaction
+    pub order: u64,
```
Comment (tdelabro) [inline src/transaction.rs:495]: I think the correct design is to create a struct 
```rust
pub struct OrderedEvent {
   content: EventContent,
   order: u64
}

File: src/transaction.rs [hunk 2]
```diff
@@ -429,6 +429,8 @@ pub struct MessageToL1 {
     pub from_address: ContractAddress,
     pub to_address: EthAddress,
     pub payload: L2ToL1Payload,
+    /// The order of the message within the transaction
+    pub order: u64,
```
Comment (tdelabro) [inline src/transaction.rs:444]: Same here. Create an `OrderedMessageToL1` struct

**Review by tdelabro — CHANGES_REQUESTED** [review]

**Comment by antiyro** [comment]
So what you should do @ArturVargas is to implement `OrderedEvent` and message directly in madara respecting starknet rs

File: src/transaction.rs (hunk 1 above)
Comment (ArturVargas) [reply src/transaction.rs:495]: I made the update, sorry for the misunderstood

**Comment by tdelabro** [comment]
Also are you aware that OrderedEvent already exists in blockifier?

https://github.com/starkware-libs/blockifier/blob/ce28997a7d695b304aa048e3783bbca8ed2b6271/crates/blockifier/src/execution/call_info.rs#L25

**Comment by ArturVargas** [comment]
Hi @tdelabro could you give me more context about where I have to implement this on Madara?

**Comment by tdelabro** [comment]
If you need it in a single place, create the struct where you need them.
If the type is needed in multiple crates, create it under `primitives`

**Pull request closed without merging**